extensions/objects/models/objects.py

```diff
@@ -1838,3 +1603,109 @@ def get_schema(cls):
         }
+
+
+class BaseTranslatableObject(BaseObject):
+    """Base translatable object class.
+
+    This is a superclass for objects that are translatable and thus require a
+    content id. This class enforces that the object is a dictionary with a
+    content id field. The schema of the actual value is determined by the
+    _value_schema property.
+    """
+
+    # The key name in the translatable object corresponding to the translatable
+    # value. This field must be populated by subclasses.
+    _value_key_name = None
+    # The schema of the translatable value. This field must be populated by
+    # subclasses.
+    _value_schema = None
+    # The default value of the object. This field must be populated by
+    # subclasses.
+    default_value = None
+
+    @classmethod
+    def normalize_value(cls, value):
+        """Normalizes the translatable value of the object.
+
+        Args:
+            value: *. The translatable part of the Python object (corresponding
+                to the non-content-id field) which is to be normalized.
+
+        Returns:
+            *. The normalized value.
+        """
+        if cls._value_key_name is None or cls._value_schema is None:
+            raise NotImplementedError(
+                'The _value_key_name and _value_schema for this class must '
+                'both be set.')
+        return schema_utils.normalize_against_schema(value, cls._value_schema)
+
+    @classmethod
+    def get_schema(cls):
+        """Returns the full object schema.
+
+        Returns:
+            dict. The object schema.
+        """
+        if cls._value_key_name is None or cls._value_schema is None:
+            raise NotImplementedError(
+                'The _value_key_name and _value_schema for this class must '
+                'both be set.')
+        return {
+            'type': 'dict',
+            'properties': [{
+                'name': 'contentId',
+                # The default content id is none. However, it should be
+                # populated before being saved. The normalize() method has
+                # validation checks for this.
+                'schema': {'type': 'unicode'}
+            }, {
+                'name': cls._value_key_name,
+                'schema': copy.deepcopy(cls._value_schema),
+            }]
+        }
+
+
+class TranslatableUnicodeString(BaseTranslatableObject):
+    """Class for translatable unicode strings."""
+
+    _value_key_name = 'unicodeStr'
+    _value_schema = UnicodeString.get_schema()
+    default_value = {
+        'contentId': None,
+        'unicodeStr': '',
+    }
+
+
+class TranslatableHtml(BaseTranslatableObject):
+    """Class for translatable HTML strings."""
+
+    _value_key_name = 'html'
+    _value_schema = Html.get_schema()
+    default_value = {
+        'contentId': None,
+        'html': '',
+    }
+
+
+class TranslatableSetOfNormalizedString(BaseTranslatableObject):
+    """Class for translatable sets of NormalizedStrings."""
```

Comment:
not sure whether it should be sets or set?

If by `NormalizedStrings` you want to highlight an existing object class then we have `SetOfNormalizedString`